data/mods/gen2/statuses.js

```diff
@@ -226,6 +223,31 @@ let BattleStatuses = {
 			this.effectData.duration = 2;
 		},
 	},
+	residualdmg: {
+		name: 'residualdmg',
+		id: 'residualdmg',
+		num: 0,
+		onStart(target) {
+			target.volatiles['residualdmg'].counter = 0;
+		},
+		onAfterMoveSelfPriority: 100,
+		onAfterMoveSelf(pokemon) {
+			pokemon.volatiles['residualdmg'].counter++;
+		},
+	},
 };
 
+/**
+ * @param {Battle} battle
+ * @param {Pokemon} pokemon
+ */
+function residualdmg(battle, pokemon) {
+	if (pokemon.volatiles['residualdmg']) {
+		battle.damage(battle.clampIntRange(Math.floor(pokemon.maxhp / 16) * pokemon.volatiles['residualdmg'].counter, 1), pokemon);
+		battle.add('-hint', 'In GSC, Toxic\'s counter is retained through Baton Pass/Heal Bell and applies to PSN/BRN.');
```

Comment:
Actually, not sure about this hint - we probably only want to show it the first time this happens, not every turn of residual damage.

Reply:
Added logic such that we only `-hint` once about this per battle (as well as created the `hint` method on Battle, which also supports one-sided hints which will be useful for #5242 or #5222).